File: frontend/src/app/services/ParcoursService.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { Parcours } from '../models/parcours.model';

@Injectable({
  providedIn: 'root'
})
export class ParcoursService {

  private apiUrl = 'http://localhost:8081/api/parcours';

  constructor(private http: HttpClient) {}

  /* GET - Récupérer tous les parcours */
  getAllParcours(): Observable<Parcours[]> {
    return this.http.get<Parcours[]>(this.apiUrl);
  }

  getParcoursById(id: number): Observable<Parcours>
  {
    return this.http.get<Parcours>(`${this.apiUrl}/${id}`);
  }

  getParcoursByClient(clientId: string) : Observable<Parcours[]> {
    return this.http.get<Parcours[]>(`${this.apiUrl}/client/${clientId}`);
  }

  getParcoursByStatus(status: string): Observable<Parcours[]> {
    return this.http.get<Parcours[]>(`${this.apiUrl}/status/${status}`);
  }

  
  /* GET - Récupérer les parcours par journeyType */
  getParcoursByJourneyType(journeyType: string): Observable<Parcours[]> {
    return this.http.get<Parcours[]>(`${this.apiUrl}/type/${journeyType}`);
  }

  /* GET - Récupérer les parcours par channel */
  getParcoursByChannel(channel: string): Observable<Parcours[]> {
    return this.http.get<Parcours[]>(`${this.apiUrl}/channel/${channel}`);
  }

  /* POST - Créer un parcours */
  createParcours(parcours: Parcours): Observable<Parcours> {
    return this.http.post<Parcours>(this.apiUrl, parcours);
  }

  /* PUT - Interrompre un parcours */
  interruptParcours(id: number): Observable<Parcours> {
    return this.http.put<Parcours>(`${this.apiUrl}/${id}/interrupt`, {});
  }

  /* PUT - Reprendre un parcours */
  resumeParcours(id: number): Observable<Parcours> {
    return this.http.put<Parcours>(`${this.apiUrl}/${id}/resume`, {});
  }

  /* PUT - Terminer un parcours */
  completeParcours(id: number): Observable<Parcours> {
    return this.http.put<Parcours>(`${this.apiUrl}/${id}/complete`, {});
  }


}

File: Backend-bff/src/main/java/com/omnitrack/service/ParcoursService.java
package com.omnitrack.service;

import com.omnitrack.client.BackendApiClient;
import com.omnitrack.dto.ParcoursDTO;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public class ParcoursService {

    @Autowired
    private BackendApiClient backendApiClient;



        /* GET - Récupérer tous les parcours */
    public List<ParcoursDTO> getAllParcours() {
        return backendApiClient.getAllParcours();
    }

    /* GET - Récupérer un parcours par ID */
    public ParcoursDTO getParcoursById(Long id) {
        return backendApiClient.getParcoursById(id);
    }

    /* GET - Récupérer les parcours par client */
    public List<ParcoursDTO> getParcoursByClientId(String clientId) {
        return backendApiClient.getParcoursByClient(clientId);
    }

    /* GET - Récupérer les parcours par status */
    public List<ParcoursDTO> getParcoursByStatus(String status) {
        return backendApiClient.getParcoursByStatus(status);
    }

    /* GET - Récupérer les parcours par journeyType */
    public List<ParcoursDTO> getParcoursByJourneyType(String journeyType) {
        return backendApiClient.getParcoursByJourneyType(journeyType);
    }

    /* GET - Récupérer les parcours par channel */
    public List<ParcoursDTO> getParcoursByChannel(String channel) {
        return backendApiClient.getParcoursByChannel(channel);
    }

    /* POST - Créer un parcours */
    public ParcoursDTO createParcours(ParcoursDTO parcoursDTO) {
        return backendApiClient.createParcours(parcoursDTO);
    }

    /* PUT - Interrompre un parcours */
    public ParcoursDTO interruptParcours(Long id) {
        return backendApiClient.interruptParcours(id);
    }

    /* PUT - Reprendre un parcours */
    public ParcoursDTO resumeParcours(Long id) {
        return backendApiClient.resumeParcours(id);
    }

    /* PUT - Terminer un parcours */
    public ParcoursDTO completeParcours(Long id) {
        return backendApiClient.completeParcours(id);
    }

}